src/mlpack/core/util/io.hpp

```diff
@@ -285,6 +285,11 @@ class IO
    */
   static void ClearSettings();
 
+  /**
+   * Checks all input matrices for NaN and inf values, if found throws an exception.
```

Comment:
Log::Fatal will not throw an error instead it will exit immediately with a return code `!= 0`.